tests/unit/mainwindow/statusbar/test_url.py

```diff
@@ -0,0 +1,126 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""Test Statusbar url."""
+
+import pytest
+
+from collections import namedtuple
+from qutebrowser.mainwindow.statusbar.url import UrlText
+from qutebrowser.utils import usertypes
+
+
+UrlType = usertypes.enum('UrlType', ['success', 'success_https', 'error',
+                                    'warn', 'hover', 'normal'])
+
+
+@pytest.fixture
+def tab_widget():
+    """Fixture providing a fake tab widget."""
+    tab = namedtuple('Tab', 'cur_url load_status')
+    tab.load_status = namedtuple('load_status', 'name')
+    tab.cur_url = namedtuple('cur_url', 'toDisplayString')
+    return tab
+
+
+@pytest.fixture
+def url_widget(qtbot, monkeypatch, config_stub):
+    """Fixture providing a Url widget."""
+    config_stub.data = {
+        'colors': {
+            'statusbar.url.bg': 'white',
+            'statusbar.url.fg': 'black',
+            'statusbar.url.fg.success': 'yellow',
+            'statusbar.url.fg.success.https': 'green',
+            'statusbar.url.fg.error': 'red',
+            'statusbar.url.fg.warn': 'orange',
+            'statusbar.url.fg.hover': 'blue'},
+        'fonts': {},
+    }
+    monkeypatch.setattr(
+        'qutebrowser.mainwindow.statusbar.url.style.config', config_stub)
+    widget = UrlText()
+    qtbot.add_widget(widget)
+    assert not widget.isVisible()
+    return widget
+
+
+@pytest.mark.parametrize('url', [
+    ('http://abc123.com/this/awesome/url.html'),
+    ('https://supersecret.gov/nsa/files.txt'),
+    ('Th1$ i$ n0t @ n0rm@L uRL! P@n1c! <-->'),
+    (None)
+])
```

Comment:
No need for the parens here, as you're only parametrizing a single argument anyways